src/octotree.js

```diff
@@ -167,28 +182,36 @@ $(document).ready(() => {
       errorView.show(err);
     }
 
-    function toggleSidebarAndSave() {
-      store.set(STORE.SHOWN, !isSidebarVisible(), () => {
-        toggleSidebar();
-        if (isSidebarVisible()) {
-          tryLoadRepo();
-        }
-      });
-    }
-
     function toggleSidebar(visibility) {
       if (visibility !== undefined) {
         if (isSidebarVisible() === visibility) return;
         toggleSidebar();
       } else {
         $html.toggleClass(SHOW_CLASS);
         $document.trigger(EVENT.TOGGLE, isSidebarVisible());
+        if (isSidebarVisible()) {
+          tryLoadRepo();
+        }
+      }
+    }
+
+    function togglePinAndSave() {
+      store.set(STORE.PINNED, !isSidebarPinned(), togglePin);
+    }
+
+    function togglePin(isPinned) {
```

Comment:
Is there a case when we toggle pin but doesn't save? If not, maybe move this logic inside togglePinAndSave (which can be rename to togglePin).